frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx

```diff
@@ -50,88 +58,132 @@ const FieldsAndSoilComponent: React.FC<InputModuleProps> = ({ farmDetails, updat
-            </StyledListContainer>
-          </StyledFarmInfo>
-        </Form>
-      )}
-    </Formik>
+            </StyledCommentContainerDesktop>
+            <StyledFontAwesomeContainer>
+              <FontAwesomeIcon icon={faPencil} />
+              <FontAwesomeIcon icon={faTrash} />
+            </StyledFontAwesomeContainer>
+          </StyledListContainer>
+          <StyledCommentContainerMobile>
+            <StyledListItem width="100%">
+              <h4>Field Comments (optional)</h4>
+              <p>{fieldsInfo.Fields[fieldIndex].Comments}</p>
+            </StyledListItem>
+          </StyledCommentContainerMobile>
+          {!fieldAdd ? (
+            <Button
+              type="button"
+              size="md"
+              disabled={false}
+              radius="50px"
+              actions="secondary"
+              text={ComponentText.NEWFIELD}
+              handleClick={addNewField}
+            >
+              <FontAwesomeIcon icon={faPlus} />
+            </Button>
+          ) : null}
+        </StyledFieldInfoList>
+      ) : null}
+      {fieldAdd || !isSubmitted ? (
+        <Formik
+          initialValues={initialValues}
+          validationSchema={validationSchema}
+          onSubmit={onSubmit}
+        >
+          {({ values }) => (
+            <Form>
+              <StyledFarmInfo>
+                <div id="inputContainer">
+                  <CustomField
+                    label="Field name"
+                    id="FieldName"
+                    name="FieldName"
+                    type="text"
+                  />
+                  <CustomField
+                    label="Area"
+                    id="Area"
+                    name="Area"
+                    type="number"
+                    width="20%"
+                  />
+                </div>
+                <StyledTextAreaContainer>
+                  <CustomTextArea
+                    name="Comments"
+                    id="Comments"
+                    label="Comments (optional)"
+                    placeholder="e.g., poor drainage in southwest corner (no need to specify crop here)"
+                    width="50%"
+                  />
+                  <StyledButtonGroupContainer>
+                    <Button
+                      type="reset"
+                      size="sm"
+                      disabled={false}
+                      actions="secondary"
+                      text={ComponentText.CANCEL}
+                    />
+                    <Button
+                      type="button"
+                      size="sm"
+                      disabled={false}
+                      text={ComponentText.ADD}
+                      handleClick={() => addFarmInfo(values)}
+                    />
+                  </StyledButtonGroupContainer>
+                </StyledTextAreaContainer>
+              </StyledFarmInfo>
+            </Form>
+          )}
+        </Formik>
+      ) : null}
```

Comment:
Another ternary that could just be an &&

Reply:
This is good to know. Idk about this syntax. 😛